app/models/participant.rb

```diff
@@ -40,6 +40,16 @@ class Participant < ApplicationRecord
   scope :for_start_list, -> do
     includes(:event).with_payment.not_cancelled.not_archived.order(:first_name)
   end
+  scope :incomplete_registration, -> do
+    joins(:registration).where(registrations: { completed_at: nil })
+  end
+  scope :last_thirty_days, -> do
+    joins(:registration).where(registrations: { started_at: 1.month.ago..Time.now })
+  end
+  scope :over_twenty_four_hours, -> do
+    joins(:registration).where("started_at <= ?", 24.hours.ago)
+  end
+  scope :abandoned_registration, -> { not_archived.incomplete_registration.not_cancelled.last_thirty_days.over_twenty_four_hours }
```

Comment:
Add tests

Reply:
Added tests to PR #189, I'm guessing I would rebase over master in chronological order and then this would have that test coverage. If I'm understanding correctly...